components/brave_search/browser/brave_search_default_host.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_BRAVE_SEARCH_BROWSER_BRAVE_SEARCH_DEFAULT_HOST_H_
+#define BRAVE_COMPONENTS_BRAVE_SEARCH_BROWSER_BRAVE_SEARCH_DEFAULT_HOST_H_
+
+#include <memory>
+#include <string>
+
+#include "base/memory/weak_ptr.h"
+#include "brave/components/brave_search/common/brave_search_default.mojom.h"
+
+class TemplateURLService;
+class TemplateURL;
+class PrefService;
+
+namespace brave_search {
+
+class BraveSearchDefaultHost final
+    : public brave_search::mojom::BraveSearchDefault {
+ public:
+  BraveSearchDefaultHost(const BraveSearchDefaultHost&) = delete;
+  BraveSearchDefaultHost& operator=(const BraveSearchDefaultHost&) = delete;
+
+  BraveSearchDefaultHost(const std::string& host,
+                         TemplateURLService* template_url_service,
+                         PrefService* prefs);
+  ~BraveSearchDefaultHost() override;
+
+  // brave_search::mojom::BraveSearchDefault:
+  void GetCanSetDefaultSearchProvider(
```

Comment:
How this works if user installed search provider extension such as(https://chrome.google.com/webstore/detail/ecosia-the-search-engine/eedlgdlajadkbbjoobobefphmfkcchfk)?
IMO, if user installed this kind of extension, maybe we should not change default search provider?

Reply:
also group policy via `is_default_search_managed()`

Reply:
fixed